coins/monero/src/wallet/seed/polyseed.rs

```diff
@@ -179,6 +179,31 @@ fn valid_entropy(entropy: &Zeroizing<[u8; 32]>) -> bool {
   res.into()
 }
 
+fn from_internal(
```

Comment:
Is there a reason to have this outside the Polyseed impl?

Reply:
`from_string` will first decode the string and yield the already masked features before passing it over to `from_internal`. The features shouldn't be masked again (the primary bug this PR is fixing).

`from` expects an *unmasked* `features`, which it then masks for the caller. I matched `polyseed_create` from the polyseed spec for this behavior, which also takes in unmasked `features` and masks it for the caller. Source: https://github.com/tevador/polyseed/blob/b7c35bb3c6b91e481ecb04fc235eaff69c507fa1/src/polyseed.c#L61 . I figured it would make sense that monero-serai's `from` would match the polyseed spec for this behavior since it's the only exposed function that a caller can pass `features` into.

Because of the above 2 distinct call paths, I made the `from_internal` take in the *already masked* features.

Sort of similar line of thinking for the birthday btw (`from_string` yields the encoded birthday so it's unnecessary to decode it, whereas `from` expects the decoded birthday which needs to be encoded).